eodag/plugins/authentication/openid_connect.py

```diff
@@ -191,11 +257,37 @@ def authenticate(self) -> CodeAuthorizedAuth:
                     tb.format_exc()
                 )
             )
-        return CodeAuthorizedAuth(
-            token,
-            self.config.token_provision,
-            key=getattr(self.config, "token_qs_key", None),
+        return token_response.json()
+
+    def get_token_with_refresh_token(self) -> Dict[str, str]:
+        """Fetch the access token with the refresh token"""
+        logger.debug(
+            "Fetching access token with refresh token from %s", self.config.token_uri
         )
+        token_data: Dict[str, Any] = {
+            "refresh_token": self.token_info["refresh_token"],
+            "grant_type": "refresh_token",
+        }
+        token_data = self._prepare_token_post_data(token_data)
+        post_request_kwargs: Any = {
+            self.config.token_exchange_post_data_method: token_data
+        }
+        try:
+            token_response = self.session.post(
+                self.config.token_uri,
+                timeout=HTTP_REQ_TIMEOUT,
+                **post_request_kwargs,
+            )
+            token_response.raise_for_status()
+        except requests.exceptions.Timeout as exc:
+            raise TimeOutError(exc, timeout=HTTP_REQ_TIMEOUT) from exc
+        except Exception as exc:
+            logger.error(
+                "Could not fetch access token with refresh token, executing new token request, error: %s",
+                getattr(exc.response, "text", ""),
+            )
+            return self.request_new_token()
+        return token_response.json()
```

Comment:
In `get_token_with_refresh_token`, please:

- [ ] get `token_time` and `refresh_time` in this method, just before sending the request
- [ ] build `refresh_token_expiration_time` and `access_token_expiration_time` from this method to make comparisons at the beginning of authenticate easier